innkeeper/client.go

```diff
@@ -320,6 +318,43 @@ func getHttpError(r *http.Response) (error, bool) {
 	}
 }
 
+func (c *Client) writeRoute(url string, route *routeData) error {
+
+	res, err := json.Marshal(route)
+
+	req, err := http.NewRequest("POST", url, bytes.NewReader(res))
+
+	if err != nil {
+		return err
+	}
+
+	authToken, err := c.opts.Authentication.GetToken()
```

Comment:
wouldn't it be better to cache the token? like in case of `requestData`? because this may mean to call the oauth service on every single route being uploaded.